packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -438,168 +445,122 @@ describe('TokenDetectionController', () => {
     expect(tokensController.state.detectedTokens).toStrictEqual([sampleTokenA]);
   });
 
-  it('should not call getBalancesInSingleCall after stopping polling, and then switching between networks that support token detection', async () => {
-    const polygonDecimalChainId = '137';
-    nock(TOKEN_END_POINT_API)
-      .get(getTokensPath(toHex(polygonDecimalChainId)))
-      .reply(200, sampleTokenList);
-
-    const stub = sinon.stub();
-    const getBalancesInSingleCallMock = sinon.stub();
-    // TODO: Replace `any` with type
-    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    let networkStateChangeListener: (state: any) => void;
-    const onNetworkStateChange = sinon.stub().callsFake((listener) => {
-      networkStateChangeListener = listener;
+  describe('getBalancesInSingleCall', () => {
```

Comment:
If we have a `describe` called this, I would expect these tests to call `getBalancesInSingleCall` directly, but they seem to involve this method through some other means. In other words this method is not the "subject" of these tests, but the "object".

I see that you are re-registering `NetworkController:findNetworkClientIdByChainId` and `NetworkController:findNetworkClientIdByChainId` here. Do you have to do this, since they're registered in the `beforeEach` at the top of this file?

If this is too difficult to change, no worries.

Reply:
The action handlers need to be registered because we're overwriting the `controllerMessenger` variable with a new instance of `MainControllerMessenger` ([line 458](https://github.com/MetaMask/core/pull/3609/files/c5eec2517fc0f981fd64cca6a3cf5d4dc7b0ba76#diff-4f72f1868a531f11ff40f993837f9cb1a1dcef7256f50f4067e91a18ed1e6040R458)). So the handlers aren't being re-registered. They're being registered for the first time to this new `controllerMessenger` instance.

For previous context on why we need a new `controllerMessenger` instance, see:
- https://github.com/MetaMask/core/pull/3609#discussion_r1424638306
- https://github.com/MetaMask/core/pull/3609#discussion_r1422856728

Reply:
Edited test descriptions so that `getBalancesInSingleCall` is the subject of the tests: https://github.com/MetaMask/core/pull/3609/commits/8c331fac7bf8b370c9f1a7b046298b7db1ba49e4